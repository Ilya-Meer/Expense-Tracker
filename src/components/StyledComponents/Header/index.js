import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const StyledNav = styled.nav`
  width: 100%;
  padding: 2rem 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  text-align: center;
  background: linear-gradient(to right, #0575e6, #021b79);
  box-shadow: 0 1px 6px rgba(0,0,0,.16), 0 1px 6px rgba(0,0,0,.23);
  position: relative;
  
  a {
    display: flex;
    align-items: center;
    text-decoration: none;
  }
`
const StyledList = styled.ul`
  width: 100%;
  margin: 0;
  padding: 3rem 0;
  display: ${props => props.show === true ? 'flex' : 'none'};
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 115px;
  right: 0;
  background: #fff;
  @media all and (min-width: 650px) {
    width: 50%;
    margin: 0 2rem;
    padding: 0;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    position: static;
    background: none;
  }

  li, a {
    color: #176dff;
    cursor: pointer;
    @media all and (min-width: 650px) {
      color: #fff;
    }
  }

  li {
    width: 150px;
    margin: 1rem 0;
    display: flex;
    list-style: none;
    @media all and (min-width: 650px) {
      width: auto;
      margin: 0;
      align-items: center;
    }
  }
`
const HomeButton = styled(NavLink)`
  width: 140px; 
  margin: 0 1rem;
  display: flex;
  justifyContent: space-between;
  color: #fff;
`
const Hamburger = styled.button`
  display: block;
  color: #fff;
  cursor: pointer;
  background: none;
  border: none;
  box-shadow: none;
  font-size: 2.5rem;
  @media all and (min-width: 650px) {
    display: none;
  }
`

export { HomeButton, Hamburger, StyledNav, StyledList };