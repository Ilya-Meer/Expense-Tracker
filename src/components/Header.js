import React, {Component} from 'react';
import { StyledNav, StyledList, Hamburger, HomeButton } from './StyledComponents/Header';
import { MdHome, MdMenu, MdPowerSettingsNew } from 'react-icons/md';
import { FaGithub } from 'react-icons/fa';
import { connect } from 'react-redux';
import { startLogout } from '../actions/authActions';

class Header extends Component {  
  constructor(props) {
    super(props);
    this.logOut = this.logOut.bind(this);
    this.toggleMobileNav = this.toggleMobileNav.bind(this);

    this.state = {
      showNav: false
    }
  }
  logOut() {
    this.props.startLogout();
  }

  toggleMobileNav() {
    this.setState({ showNav: !this.state.showNav })
  }

  render() {
    return (
        <StyledNav>
          <HomeButton
            to="/dashboard">
              <MdHome/>Exp3nse Tracker
          </HomeButton>
          <Hamburger 
            onClick={this.toggleMobileNav}> 
              <MdMenu />
          </Hamburger>
          <StyledList show={this.state.showNav}>
            <li>
              <a
                href="https://github.com/Ilya-Meer/Expense-Tracker"
                target="_blank"
                rel="noopener">
                  <FaGithub/>View on Github
              </a>
            </li>
            <li onClick={this.logOut}>
              <MdPowerSettingsNew/>
              Log Out
            </li>
          </StyledList>
        </StyledNav>
    )
  }
}

export { Header };

const mapDispatchToProps = (dispatch) => {
  return {
    startLogout: () => {
      dispatch(startLogout())
    }
  }
}

export default connect(undefined, mapDispatchToProps)(Header)

 