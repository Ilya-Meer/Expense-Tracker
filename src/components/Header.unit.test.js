import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header Component', () => {
  const wrapper = shallow(<Header startLogout={() => {}}/>);

  it('renders properly', () => {  
    expect(wrapper).toMatchSnapshot();
  })


  it('logs user out when logout button is clicked', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
  })
})
