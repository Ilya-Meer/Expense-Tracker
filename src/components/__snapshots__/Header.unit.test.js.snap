// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders properly 1`] = `
<styled.nav>
  <Styled(NavLink)
    activeClassName="selected"
    activeStyle={
      Object {
        "color": "#bcddff",
        "textDecoration": "none",
      }
    }
    to="/dashboard"
  >
    <Component />
    Exp3nse Tracker
  </Styled(NavLink)>
  <styled.button
    onClick={[Function]}
  >
    <Component />
  </styled.button>
  <styled.ul
    show={false}
  >
    <li>
      <NavLink
        activeClassName="selected"
        activeStyle={
          Object {
            "color": "#bcddff",
            "textDecoration": "none",
          }
        }
        aria-current="page"
        to="/create"
      >
        <Component />
        Add Expense
      </NavLink>
    </li>
    <li>
      <a
        href="https://github.com/Ilya-Meer/Expense-Tracker"
        rel="noopener"
        target="_blank"
      >
        <Component />
        View on Github
      </a>
    </li>
    <li
      onClick={[Function]}
    >
      <Component />
      Log Out
    </li>
  </styled.ul>
</styled.nav>
`;
